blockdag/reachability.go

```diff
@@ -382,6 +617,23 @@ func (rtn *reachabilityTreeNode) isAncestorOf(other *reachabilityTreeNode) bool
 	return rtn.interval.isAncestorOf(other.interval)
 }
 
+// findCommonAncestor finds the most recent reachability tree ancestor
+// common to both rtn and other.
+func (rtn *reachabilityTreeNode) findCommonAncestor(other *reachabilityTreeNode) *reachabilityTreeNode {
+	currentThis := rtn
+	currentOther := other
+	for {
+		if currentThis.parent == nil || currentThis.isAncestorOf(other) {
```

Comment:
`currentThis.parent == nil` implies that currentThis is genesis, right?
But in that case `currentThis.isAncestorOf(other)` will also be true, wouldn't it?

In that case I'd remove the `currentThis.parent == nil` condition.

Then when we get to `currentThis = currentThis.parent` and one of the guys has no parents - panic :fearful: 

Why here I'm OK with panicing while in other cases I was against (it's still a shift in my position, but please don't make it a slippery slope) - this is an assumption that the reachability module has - all tree nodes have a common ancestor, unlike most cases where we wanted to assert a more general assumption in a specific library.
I'd still rather return an error in that case, but that will propagate error-returning to a ton of functions, so I'm fine with panicing this one time.